spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderHealthIndicator.java

```diff
@@ -77,57 +73,38 @@
 
 	@Override
 	protected void doHealthCheck(Health.Builder builder) throws Exception {
-		try {
-			initAdminClient();
-			synchronized (this.adminClient) {
-				final Status status = healthStatusThreadLocal.get();
-				//If one of the kafka streams binders (kstream, ktable, globalktable) was down before on the same request,
-				//retrieve that from the thead local storage where it was saved before. This is done in order to avoid
-				//the duration of the total health check since in the case of Kafka Streams each binder tries to do
-				//its own health check and since we already know that this is DOWN, simply pass that information along.
-				if (status == Status.DOWN) {
-					builder.withDetail("No topic information available", "Kafka broker is not reachable");
-					builder.status(Status.DOWN);
-				}
-				else {
-					final ListTopicsResult listTopicsResult = this.adminClient.listTopics();
-					listTopicsResult.listings().get(this.configurationProperties.getHealthTimeout(), TimeUnit.SECONDS);
-
-					if (this.kafkaStreamsBindingInformationCatalogue.getStreamsBuilderFactoryBeans().isEmpty()) {
-						builder.withDetail("No Kafka Streams bindings have been established", "Kafka Streams binder did not detect any processors");
-						builder.status(Status.UNKNOWN);
-					}
-					else {
-						boolean up = true;
-						for (KafkaStreams kStream : kafkaStreamsRegistry.getKafkaStreams()) {
-							up &= kStream.state().isRunning();
-							builder.withDetails(buildDetails(kStream));
-						}
-						builder.status(up ? Status.UP : Status.DOWN);
+		try (AdminClient adminClient = AdminClient.create(this.adminClientProperties)) {
```

Comment:
This still has the problem of creating `AdminClient` on each request. We will fix it when we merge the PR.

Reply:
That is true. FYO: I have noticed this `synchronized (this.adminClient)`. Synchronizing on non-final is not recommended.
I didn't know if `AdminClient` is capable of reconnection. If it does not - there should be a way to determine if needs to be recreated/reconnected.  
This object looks very expensive as starts a new tread every time  (that can be used as a potential attack vector).  Another point to consider - caching status for some time (even for 100ms).